software/teachee-desktop/src/app.rs

```diff
@@ -82,29 +112,143 @@ impl eframe::App for App {
+                        ui.label("Channel scaling");
+
+                        ui.add_space(GROUP_SPACING);
+                        ui.separator();
+                        ui.add_space(GROUP_SPACING);
+
+                        ui.label("Horizontal");
+                        ui.columns(2, |uis| {
+                            uis[0].label("Offset");
+                            uis[0].add(Slider::new(h_offset, 0.0..=100.0).show_value(false));
+                            uis[1].label("Scale");
+                            uis[1].add(Slider::new(h_scale, 0.0..=100.0).show_value(false));
+                        });
+
+                        ui.add_space(GROUP_SPACING);
+                        ui.separator();
+                        ui.add_space(GROUP_SPACING);
+
+                        ui.label("Vertical");
+                        ui.group(|ui| {
+                            ui.vertical_centered_justified(|ui| {
+                                // TODO: save vertical offset/scale when checked and
+                                // update individual channel's vertical offset/scale
+                                // when unchecked
+                                ui.checkbox(channel1_on, "Channel 1");
+                            });
+                        });
+                        ui.group(|ui| {
+                            ui.vertical_centered_justified(|ui| {
+                                ui.checkbox(channel2_on, "Channel 2");
+                            });
+                        });
+                        ui.columns(2, |uis| {
+                            uis[0].label("Offset");
+                            uis[0].add(Slider::new(v_offset, 0.0..=100.0).show_value(false));
+                            uis[1].label("Scale");
+                            uis[1].add(Slider::new(v_scale, 0.0..=100.0).show_value(false));
+                        });
+
+                        ui.add_space(GROUP_SPACING);
+                        ui.separator();
+                        ui.add_space(GROUP_SPACING);
+
+                        ui.vertical_centered_justified(|ui| {
+                            if ui
+                                .add(Button::new("Reset").min_size((0.0, BUTTON_HEIGHT).into()))
+                                .clicked()
+                            {
+                                *h_offset = 0.0;
+                                *h_scale = 0.0;
+                                *v_offset = 0.0;
+                                *v_scale = 0.0;
+                                *saved_v_offset = 0.0;
+                                *saved_v_scale = 0.0;
+                                *channel1_v_offset = 0.0;
+                                *channel1_v_scale = 0.0;
+                                *channel2_v_offset = 0.0;
+                                *channel2_v_scale = 0.0;
+                                *channel1_on = false;
+                                *channel2_on = false;
```

Comment:
This is basically hardcoding `self = Default();` for a subset of the fields on `self`. It might make more sense to split these fields into a new struct so we can reset them all at once using something like `app.scope_ui = Default();`.

Reply:
Can't do `*self = Self::default();` since we're mutably borrowing self's fields twice (one at `let Self {} = self;` and the other when setting to default). So I assume if we destructure then we can't access through self.

Reply:
See latest commit. We either destructure everything but manually assign in reset or keep UIControls but have to litter the code with `&mut ui_controls.<field>`